smithy-codegen-core/src/test/java/software/amazon/smithy/codegen/core/TraceFileTest.java

```diff
@@ -0,0 +1,267 @@
+package software.amazon.smithy.codegen.core;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import software.amazon.smithy.model.node.ExpectationNotMetException;
+import software.amazon.smithy.model.shapes.ShapeId;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+
+class TraceFileTest {
+    /* Correct TraceFile use tests
+    -Test creating, parsing, validating and writing a TraceFile from scratch.
+    -Test parse and write functionality combined on a correct trace file
+    -Test parse and write functionality with nodes with single children instead of arrays of children
+    -Test parse and write functionality combined on a correct trace file without the optional definitions section
+    */
+
+    @Test
+    void assertWriteTraceFileFromScratchWorks() throws IOException, URISyntaxException {
+        /*
+        Building ArtifactMetadata - this builder uses setTimestampAsNow, but you can also use a different
```

Comment:
nit: multi-line comments should get formatted like this:
```suggestion
        /**
         * Building ArtifactMetadata - this builder uses setTimestampAsNow, but you can also use a different
```